src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
> though we should also extend it to tail calls

At a closer look, tail calls already do the correct thing, as they mark all arg registers as in-use before generating the GS check and free them after generating the call. In view of that behavior, I am now entirely sure that the proposed changes are the appropriate solution.